### PR TITLE
Felinid Thievery

### PR DESCRIPTION
**Pull request by WarMechanic**
## About the PR

This PR updates Felinids with a soft rework to their old passive thieving gloves ability. Now, Felinids have instead learned to see hidden items when stripping other mobs (i.e. items in pockets) and strip items faster in general (100% -> 66.67%). Admin Observers get this ability too.

## Why / Balance

Since the removal of passive thieving gloves (see https://github.com/DeltaV-Station/Delta-v/pull/666), Felinids cannot so much be called pickpockets anymore since they get little to their name that benefits such. Allowing Felinids to see into players' pockets and increasing stripping speed are parts of my suggested changes to work the pickpocket identity back into the Felinid kit while having implications for both antags and non-antags.

> Weren't passive thieving gloves removed for good reason? / Won't this change promote shittery? / Any other variation thereof.

Passive stealth was indeed removed for a very sensible and good reason, but I would like to make the distinction that passive stealth enabled uncontested shittery since Felinids could simply have gloves, magboots or even belts disappear with little retaliation and RP value by extension, this comparison honestly isn't fair. This PR addresses this concern by leaving out stealth and only introducing pocket sight and strip speed buffs. I tried to implement this new set of changes with non/antag viability in mind. While old Felinids' passive thieving gloves let you skip on 4tc to rob people, this new iteration doesn't hide doafter which allows for retaliation. Any attempts made to steal, irrespective of success can and will go noticed by players (unless thieving gloves are used). But you can even combine thieves gloves with Felinid strip speed as an antag Felinid for a fastest thief gimmick (has similar energy to species specific traitor items). Pocket sight can give any player useful information and faster stripping has implications in both the Security department and everyday stealing. 

At the end of the day, Felinids could likely be used by 'shitters' to steal an item or two. But remember that stripping is an action largely restricted to roleplay and non-combat scenarios. As such, this rebalance would not affect parts of the game an Oni's stat stick with accuracy penalty would (the average greytider will not use a gun, and benefits from increased crowbar/baseball damage). This buff will likely balance itself out over time to rising awareness and an anticipated 'Felinid = thief' stigma. Subversions of this stigma provides opportunity for roleplay.

> Why pocket sight?

Felinid pocket sight grants them a new way to comb information that's largely befitting of their roguish archetype. While in the past the same results could have been achieved with passive stealth, this ability creates agency regarding the value of an item and whether to actually attempt stealing it in the first place (especially since you no longer have passive stealth). Having information unique to a species also creates opportunity for 'boy who cried wolf' situations wherein a set of people decide on the merit of a Felinid's argument: "i saw an e-saber in his pocket!!!", as well as providing reason for Antags to distrust Felinids (nobody should trust Felinids).

> Why pocket sight AND strip speed buff?

While the PR could be left with just pocket sight, pockets only pertain to two item slots where one of them likely holds an emergency oxygen tank anyway. But most importantly pocket sight is likely to go unnoticed by a majority of players that do not play Felinids or look at the github / changelogs, since pocket sight rejects an established notion that you cannot see into another players' pocket. It almost feels metagamey, like an afterthought to mention it. Adding a strip speed buff is very much complementary to the pocket sight addition since it provides diegetic ground for players to acknowledge Felinids' pickpocket potential. While pocket sight only directly affects the user, faster stealing can be witnessed by all players. As for the numbers, 66.67% strip time brings down pocket/radio/shoes/hand strip times from (3 -> 2) seconds and backpack (6 -> 4) seconds. 2 seconds I've observed to be the reaction time of most players, so Felinids will be able to nip items more consistently. I chose multiplicative strip time as faster belt stripping has better implications for Security when searching for contraband. If one were to have their belt stolen, it is likely their fault for not paying attention and would have had it stolen anyway.

I talk to myself here regarding balance choices: https://discord.com/channels/968983104247185448/1228169033547776152

## Technical details

Content.Client/.../StrippableBoundUserInterface.cs checks for ThievingComponent and its IgnoreStripHidden variable on the local entity before obscuring hidden items. Because this is client-sided I do not know if this is the safest implementation of this feature.

StripTimeMultiplier variable was also added to Thieving, which simply updates a preexisting multiplier argument that's passed to an event.

## Media

image

https://youtu.be/cAVxXPUNImI?si=fBuKBQHGrzzJn03f

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- tweak: Added the IgnoreStripHidden variable to ThievingComponent, which allows entities to look into the pockets of mobs it strips.
- tweak: Added the StripTimeMultiplier variable to ThievingComponent, which multiplies strip time.
- add: Felinids and Admin Observers are now able to identify items in pockets of other players. Felinids also gain a faster strip speed.

### CODE REVIEW

File: Content.Shared/Strip/Components/ThievingComponent.cs
Comment (WarMechanic) [inline Content.Shared/Strip/Components/ThievingComponent.cs:1]: I don't know what SixLabors.ImageSharp is, or how it got there. Allow me to remove it.

File: Content.Shared/Strip/Components/ThievingComponent.cs [hunk 1]
```diff
@@ -9,14 +9,21 @@ public sealed partial class ThievingComponent : Component
     /// <summary>
     /// How much the strip time should be shortened by
     /// </summary>
-    [ViewVariables(VVAccess.ReadWrite)]
-    [DataField("stripTimeReduction")]
+    [ViewVariables(VVAccess.ReadWrite), DataField("stripTimeReduction")]
```
Comment (deltanedas) [inline Content.Shared/Strip/Components/ThievingComponent.cs]: dont touch this

File: Content.Shared/Strip/Components/ThievingComponent.cs [hunk 2]
```diff
@@ -9,14 +9,21 @@ public sealed partial class ThievingComponent : Component
...
 
+    [ViewVariables(VVAccess.ReadWrite), DataField("stripTimeMultiplier")]
+    public float StripTimeMultiplier = 1f;
+
     /// <summary>
     /// Should it notify the user if they're stripping a pocket?
     /// </summary>
-    [ViewVariables(VVAccess.ReadWrite)]
-    [DataField("stealthy")]
+    [ViewVariables(VVAccess.ReadWrite), DataField("stealthy")]
```
Comment (deltanedas) [inline Content.Shared/Strip/Components/ThievingComponent.cs]: or this

File: Content.Shared/Strip/Components/ThievingComponent.cs [hunk 3]
```diff
@@ -9,14 +9,21 @@ public sealed partial class ThievingComponent : Component
...
     public bool Stealthy;
+
+    /// <summary>
+    ///  Should the user be able to see hidden items? (i.e pockets)
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite), DataField("ignoreStripHidden")]
```
Comment (deltanedas) [inline Content.Shared/Strip/Components/ThievingComponent.cs]: replace this line with [DataField] and add deltav marker comment

File: Content.Shared/Strip/Components/ThievingComponent.cs [hunk 4]
```diff
@@ -9,14 +9,21 @@ public sealed partial class ThievingComponent : Component
     /// <summary>
     /// How much the strip time should be shortened by
     /// </summary>
-    [ViewVariables(VVAccess.ReadWrite)]
-    [DataField("stripTimeReduction")]
+    [ViewVariables(VVAccess.ReadWrite), DataField("stripTimeReduction")]
     public TimeSpan StripTimeReduction = TimeSpan.FromSeconds(0.5f);
 
+    [ViewVariables(VVAccess.ReadWrite), DataField("stripTimeMultiplier")]
+    public float StripTimeMultiplier = 1f;
```
Comment (deltanedas) [inline Content.Shared/Strip/Components/ThievingComponent.cs]: this is unused remove

File: Content.Client/Inventory/StrippableBoundUserInterface.cs [hunk 5]
```diff
@@ -31,6 +32,7 @@ namespace Content.Client.Inventory
     public sealed class StrippableBoundUserInterface : BoundUserInterface
     {
         [Dependency] private readonly IUserInterfaceManager _ui = default!;
+        [Dependency] internal readonly IPlayerManager PlayerManager = default!;
```
Comment (deltanedas) [inline Content.Client/Inventory/StrippableBoundUserInterface.cs]: private not internal, nothing else uses it

also deltav marker here and on the usage line

**Review by deltanedas — CHANGES_REQUESTED** [review]

File: Content.Shared/Strip/Components/ThievingComponent.cs (hunk 4 above)
Comment (WarMechanic) [reply Content.Shared/Strip/Components/ThievingComponent.cs]: StripTimeMultiplier I used for Felinid buff. Should I remove it and readd it in its own PR?

File: Content.Shared/Strip/Components/ThievingComponent.cs (hunk 3 above)
Comment (WarMechanic) [reply Content.Shared/Strip/Components/ThievingComponent.cs]: What is a deltav marker comment? Is it literally a comment that signifies that the line of code pertains to DeltaV?

File: Content.Shared/Strip/Components/ThievingComponent.cs (hunk 4 above)
Comment (deltanedas) [reply Content.Shared/Strip/Components/ThievingComponent.cs]: there is no felenid buff you reverted that

also felinids having thieving was removed because felenids were ultimate shitter/syndie species

File: Content.Shared/Strip/Components/ThievingComponent.cs (hunk 3 above)
Comment (deltanedas) [reply Content.Shared/Strip/Components/ThievingComponent.cs]: yes

File: Content.Shared/Strip/Components/ThievingComponent.cs (hunk 4 above)
Comment (WarMechanic) [reply Content.Shared/Strip/Components/ThievingComponent.cs]: Thieving was removed I presume because items vanishing from inventories without a trace would be frustrating which I agree is sensible. However my buff would function as a soft reintroduction of a pickpocket passive that has more implications for non-antags. I have a discussion here: https://discord.com/channels/968983104247185448/1228169033547776152

File: Content.Shared/Strip/Components/ThievingComponent.cs [hunk 6]
```diff
@@ -19,4 +19,11 @@ public sealed partial class ThievingComponent : Component
     [ViewVariables(VVAccess.ReadWrite)]
     [DataField("stealthy")]
     public bool Stealthy;
+
+    /// <summary>
+    ///  Should the user be able to see hidden items? (i.e pockets)
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite)]
+    [DataField("ignoreStripHidden")] //DeltaV
```
Comment (deltanedas) [inline Content.Shared/Strip/Components/ThievingComponent.cs]: again just [DataField] its automatically rw and uses camelCase variable name

= false can be removed thats the default

also nothing uses it so...

Comment (WarMechanic) [reply Content.Shared/Strip/Components/ThievingComponent.cs]: Can you rephrase this? I don't know what you want me to do

Comment (deltanedas) [reply Content.Shared/Strip/Components/ThievingComponent.cs]: delete those lines and have

```cs
[DataField]
public bool IgnoreStripHidden; // DeltaV
```

then make it actually used by aghost and felenid prototype, but felenids should still show doafter

**Comment by Axionyxx** [comment]
Felinid thiefing was removed on purpose

**Comment by deltanedas** [comment]
this has no reduction and still shows doafter so seems ok balance wise

**Comment by WarMechanic** [comment]
What's wrong with reduction? I believe that reduction while still maintaining doafter while allowing you to successfully take something, wouldn't let you get away with said item for free. Most people respond to pickpocketing over the halfway mark so I thought it would be a fair change.

Multiplicative reduction which I reverted earlier I implemented because multiplicative strip time reduction has use cases in Security to strip belts and bags off criminals to search for contraband faster.

Although at the end of the day I will probably put it in a separate PR so the changes are easily revertible.

**Comment by WarMechanic** [comment]
I've decided that there may have been something I misunderstood regarding the PR guidelines on Robust Book since the scope of this PR was originally going to be the tweak by itself.

I would greatly appreciate advice on how to treat this PR going forward.

**Comment by WarMechanic** [comment]
How do I fix the outdated review?

**Review by WarMechanic — COMMENTED** [review]
Testing, can this outdated review resolve?

**Review by deltanedas — APPROVED** [review]
i approve of the implementation

**Comment by WarMechanic** [comment]
holy shit

**Comment by rosieposieeee** [comment]
in my completely one hundred percent unbiased opinion, i think vulps should get this too since they lost thieving gloves as well.

**Comment by WarMechanic** [comment]
huh, skub

i never saw any remove thieving from vulps pr but yeah i looked at the first pr and thieving is there. i think pickpocketing fits with felinids because they have a small size, but it should be easy to copy and paste the thieving component from felinid to vulp if you want to pr it.

**Comment by WarMechanic** [comment]
how do i fix do not merge, is it a balance concern

**Comment by WarMechanic** [comment]
thanks a bunch deltanedas

**Comment by WarMechanic** [comment]
any word on this? slimes can turn into geras and have minor internal storage now

**Comment by therealDLondon** [comment]
personally from a balance standpoint im fine with this, Especially considering other species have been buffed somewhat recently
